### ISSUE TITLE
Find conflicting fields programatically

### ISSUE THREAD

**Issue by containscafeine** [kedgeproject/kedge#111]
We embed a lot of structs (huuuge structs), and it's very likely that there are some conflicting fields which we cannot catch just by looking.

We need to way to find conflicting fields in our resulting spec, and then handle such conflicts gracefully and very reliably.

One example is while implementing jobs (#52), I figured that the `activeDeadlineSeconds` field would not populate in the JobSpec, and @surajssd and I figured out that this field exists in both `api.PodSpec` and `batch.JobSpec` and the way we are building jobs is -

```golang
type JobSpecMod struct {
	Name          string `json:"name,omitempty"`
	api.PodSpec   `json:",inline"`
	batch.JobSpec `json:",inline"`
}
```

Hence, this field does not get populated in either of the structs due to the conflict.

In this case, the conflicting field `activeDeadlineSeconds` specifies the number of seconds a pod is allowed to run on a node, so it makes sense to substitute the same value in both `JobSpec` and `PodSpec`.

Similarly, we need to take decisions for all the conflicting fields.

So -
- [ ] Find conflicting fields
- [ ] Handle conflicting fields such that the unmarshalled data for these is not lost and is stored somewhere
- [ ] Make intelligent decision for each conflicting field based on the input data

**Comment by kadel** [comment]
if i understand it correctly activeDeadlineSeconds in Job is doing something slightly different than activeDeadlineSeconds in Pod. It should be possible to specify both separately

**Comment by kadel** [comment]
How do you propose to fix conflicting keys programmatically?

**Comment by containscafeine** [comment]
@kadel I don't think the fixing can be done programmatically :yum:, I think those checkboxes are more of the workflow we should follow. This issue can be closed in full with #135
